freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/FreeIpaUsersStateProvider.java

```diff
@@ -56,6 +68,31 @@ public UsersState getUsersState(FreeIpaClient freeIpaClient) throws FreeIpaClien
         return builder.build();
     }
 
+    private void addUserToUsersState(UsersState.Builder builder, User user) {
+        Optional<UserMetadata> userMetadata = userMetadataConverter.toUserMetadata(user);
+        builder.addUser(fromIpaUser(user, userMetadata));
+        userMetadata.ifPresent(meta -> builder.addUserMetadata(user.getUid(), meta));
+        if (null != user.getMemberOfGroup()) {
+            user.getMemberOfGroup().stream()
+                    .filter(group -> !IPA_UNMANAGED_GROUPS.contains(group))
+                    .forEach(group -> builder.addMemberToGroup(group, user.getUid()));
+        } else {
+            LOGGER.warn("User {} is not a member of any groups.", user.getUid());
+        }
+    }
+
+    private Set<User> getUsers(FreeIpaClient freeIpaClient, boolean splitUserRetrieval)  throws FreeIpaClientException {
+        if (splitUserRetrieval) {
```

Comment:
could you add log, just to see which branch is used

Reply:
done